### ISSUE TITLE
Logit caching for target and shadow models

### ISSUE THREAD

**Issue by johanos1** [aidotse/LeakPro#335]
# Feature
## Desired Behavior / Functionality
Cache logits.

## What Needs to Be Done
Implement caching of logits.
